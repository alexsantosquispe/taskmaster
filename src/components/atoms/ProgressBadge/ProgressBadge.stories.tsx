import type { Meta, StoryObj } from '@storybook/react';

import { Wrapper } from '@/utils/testing/unitTest.util';
import { ProgressBadge } from './ProgressBadge';

const meta = {
  title: 'Atoms/ProgressBadge',
  component: ProgressBadge,
  parameters: {
    layout: 'centered'
  }
} satisfies Meta<typeof ProgressBadge>;

export default meta;

type Story = StoryObj<typeof ProgressBadge>;

export const Default: Story = () => {
  return (
    <div className="flex items-center gap-4">
      <Wrapper className="gap-8">
        <ProgressBadge percentage={0} />
        <ProgressBadge percentage={40} />
        <ProgressBadge percentage={60} />
        <ProgressBadge percentage={80} />
      </Wrapper>
    </div>
  );
};

Default.args = {};
