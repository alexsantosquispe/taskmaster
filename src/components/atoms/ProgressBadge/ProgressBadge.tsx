import { getPercentageStyles } from '../../../utils';
import { CircularProgress } from '../CircularProgress/CircularProgress';

interface ProgressBadgeProps {
  percentage: number;
}

export const ProgressBadge = ({ percentage }: ProgressBadgeProps) => {
  const percentageStyles = getPercentageStyles(percentage);

  return (
    <div className="flex items-center gap-1 rounded-md border border-neutral-100 px-2 py-1 text-neutral-600 shadow-xs dark:border-neutral-700 dark:text-white/80">
      <CircularProgress percentage={percentage} className={percentageStyles} />
      <span className={percentageStyles}>{percentage}%</span>
    </div>
  );
};
