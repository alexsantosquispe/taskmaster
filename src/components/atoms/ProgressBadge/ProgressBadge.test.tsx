import { render } from '@testing-library/react';
import { ProgressBadge } from './ProgressBadge';

describe('ProgressBadge', () => {
  describe('styles', () => {
    it('should render the component correctly', () => {
      let component = render(<ProgressBadge percentage={0} />);
      expect(component).toMatchSnapshot();

      component = render(<ProgressBadge percentage={25} />);
      expect(component).toMatchSnapshot();

      component = render(<ProgressBadge percentage={60} />);
      expect(component).toMatchSnapshot();

      component = render(<ProgressBadge percentage={90} />);
      expect(component).toMatchSnapshot();
    });
  });
});
