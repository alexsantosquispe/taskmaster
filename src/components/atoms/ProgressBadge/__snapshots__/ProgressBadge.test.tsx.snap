// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`ProgressBadge styles should render the component correctly 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="flex items-center justify-center gap-1 rounded-md border border-neutral-200 px-1 py-0.5 font-semibold text-neutral-600 shadow-sm dark:border-neutral-700 dark:text-white/80"
      >
        <svg
          height="16"
          width="16"
        >
          <circle
            class="stroke-gray-200 dark:stroke-neutral-600"
            cx="8"
            cy="8"
            fill="none"
            r="7"
            stroke-width="2"
          />
          <circle
            class="stroke-primary dark:stroke-white text-neutral-600 dark:text-white/80"
            cx="8"
            cy="8"
            fill="none"
            r="7"
            stroke-dasharray="43.982297150257104"
            stroke-dashoffset="43.54247417875453"
            stroke-linecap="round"
            stroke-width="2"
            transform="rotate(-90 8 8)"
          />
        </svg>
        <span
          class="stroke-primary dark:stroke-white text-neutral-600 dark:text-white/80"
        >
          0
          %
        </span>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="flex items-center justify-center gap-1 rounded-md border border-neutral-200 px-1 py-0.5 font-semibold text-neutral-600 shadow-sm dark:border-neutral-700 dark:text-white/80"
    >
      <svg
        height="16"
        width="16"
      >
        <circle
          class="stroke-gray-200 dark:stroke-neutral-600"
          cx="8"
          cy="8"
          fill="none"
          r="7"
          stroke-width="2"
        />
        <circle
          class="stroke-primary dark:stroke-white text-neutral-600 dark:text-white/80"
          cx="8"
          cy="8"
          fill="none"
          r="7"
          stroke-dasharray="43.982297150257104"
          stroke-dashoffset="43.54247417875453"
          stroke-linecap="round"
          stroke-width="2"
          transform="rotate(-90 8 8)"
        />
      </svg>
      <span
        class="stroke-primary dark:stroke-white text-neutral-600 dark:text-white/80"
      >
        0
        %
      </span>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`ProgressBadge styles should render the component correctly 2`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="flex items-center justify-center gap-1 rounded-md border border-neutral-200 px-1 py-0.5 font-semibold text-neutral-600 shadow-sm dark:border-neutral-700 dark:text-white/80"
      >
        <svg
          height="16"
          width="16"
        >
          <circle
            class="stroke-gray-200 dark:stroke-neutral-600"
            cx="8"
            cy="8"
            fill="none"
            r="7"
            stroke-width="2"
          />
          <circle
            class="stroke-primary dark:stroke-white text-neutral-600 dark:text-white/80"
            cx="8"
            cy="8"
            fill="none"
            r="7"
            stroke-dasharray="43.982297150257104"
            stroke-dashoffset="43.54247417875453"
            stroke-linecap="round"
            stroke-width="2"
            transform="rotate(-90 8 8)"
          />
        </svg>
        <span
          class="stroke-primary dark:stroke-white text-neutral-600 dark:text-white/80"
        >
          0
          %
        </span>
      </div>
    </div>
    <div>
      <div
        class="flex items-center justify-center gap-1 rounded-md border border-neutral-200 px-1 py-0.5 font-semibold text-neutral-600 shadow-sm dark:border-neutral-700 dark:text-white/80"
      >
        <svg
          height="16"
          width="16"
        >
          <circle
            class="stroke-gray-200 dark:stroke-neutral-600"
            cx="8"
            cy="8"
            fill="none"
            r="7"
            stroke-width="2"
          />
          <circle
            class="stroke-amber-600 dark:stroke-amber-400 text-amber-700 dark:text-amber-400"
            cx="8"
            cy="8"
            fill="none"
            r="7"
            stroke-dasharray="43.982297150257104"
            stroke-dashoffset="32.98672286269283"
            stroke-linecap="round"
            stroke-width="2"
            transform="rotate(-90 8 8)"
          />
        </svg>
        <span
          class="stroke-amber-600 dark:stroke-amber-400 text-amber-700 dark:text-amber-400"
        >
          25
          %
        </span>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="flex items-center justify-center gap-1 rounded-md border border-neutral-200 px-1 py-0.5 font-semibold text-neutral-600 shadow-sm dark:border-neutral-700 dark:text-white/80"
    >
      <svg
        height="16"
        width="16"
      >
        <circle
          class="stroke-gray-200 dark:stroke-neutral-600"
          cx="8"
          cy="8"
          fill="none"
          r="7"
          stroke-width="2"
        />
        <circle
          class="stroke-amber-600 dark:stroke-amber-400 text-amber-700 dark:text-amber-400"
          cx="8"
          cy="8"
          fill="none"
          r="7"
          stroke-dasharray="43.982297150257104"
          stroke-dashoffset="32.98672286269283"
          stroke-linecap="round"
          stroke-width="2"
          transform="rotate(-90 8 8)"
        />
      </svg>
      <span
        class="stroke-amber-600 dark:stroke-amber-400 text-amber-700 dark:text-amber-400"
      >
        25
        %
      </span>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`ProgressBadge styles should render the component correctly 3`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="flex items-center justify-center gap-1 rounded-md border border-neutral-200 px-1 py-0.5 font-semibold text-neutral-600 shadow-sm dark:border-neutral-700 dark:text-white/80"
      >
        <svg
          height="16"
          width="16"
        >
          <circle
            class="stroke-gray-200 dark:stroke-neutral-600"
            cx="8"
            cy="8"
            fill="none"
            r="7"
            stroke-width="2"
          />
          <circle
            class="stroke-primary dark:stroke-white text-neutral-600 dark:text-white/80"
            cx="8"
            cy="8"
            fill="none"
            r="7"
            stroke-dasharray="43.982297150257104"
            stroke-dashoffset="43.54247417875453"
            stroke-linecap="round"
            stroke-width="2"
            transform="rotate(-90 8 8)"
          />
        </svg>
        <span
          class="stroke-primary dark:stroke-white text-neutral-600 dark:text-white/80"
        >
          0
          %
        </span>
      </div>
    </div>
    <div>
      <div
        class="flex items-center justify-center gap-1 rounded-md border border-neutral-200 px-1 py-0.5 font-semibold text-neutral-600 shadow-sm dark:border-neutral-700 dark:text-white/80"
      >
        <svg
          height="16"
          width="16"
        >
          <circle
            class="stroke-gray-200 dark:stroke-neutral-600"
            cx="8"
            cy="8"
            fill="none"
            r="7"
            stroke-width="2"
          />
          <circle
            class="stroke-amber-600 dark:stroke-amber-400 text-amber-700 dark:text-amber-400"
            cx="8"
            cy="8"
            fill="none"
            r="7"
            stroke-dasharray="43.982297150257104"
            stroke-dashoffset="32.98672286269283"
            stroke-linecap="round"
            stroke-width="2"
            transform="rotate(-90 8 8)"
          />
        </svg>
        <span
          class="stroke-amber-600 dark:stroke-amber-400 text-amber-700 dark:text-amber-400"
        >
          25
          %
        </span>
      </div>
    </div>
    <div>
      <div
        class="flex items-center justify-center gap-1 rounded-md border border-neutral-200 px-1 py-0.5 font-semibold text-neutral-600 shadow-sm dark:border-neutral-700 dark:text-white/80"
      >
        <svg
          height="16"
          width="16"
        >
          <circle
            class="stroke-gray-200 dark:stroke-neutral-600"
            cx="8"
            cy="8"
            fill="none"
            r="7"
            stroke-width="2"
          />
          <circle
            class="stroke-green-600 dark:stroke-green-400 text-green-600 dark:text-green-400"
            cx="8"
            cy="8"
            fill="none"
            r="7"
            stroke-dasharray="43.982297150257104"
            stroke-dashoffset="17.59291886010284"
            stroke-linecap="round"
            stroke-width="2"
            transform="rotate(-90 8 8)"
          />
        </svg>
        <span
          class="stroke-green-600 dark:stroke-green-400 text-green-600 dark:text-green-400"
        >
          60
          %
        </span>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="flex items-center justify-center gap-1 rounded-md border border-neutral-200 px-1 py-0.5 font-semibold text-neutral-600 shadow-sm dark:border-neutral-700 dark:text-white/80"
    >
      <svg
        height="16"
        width="16"
      >
        <circle
          class="stroke-gray-200 dark:stroke-neutral-600"
          cx="8"
          cy="8"
          fill="none"
          r="7"
          stroke-width="2"
        />
        <circle
          class="stroke-green-600 dark:stroke-green-400 text-green-600 dark:text-green-400"
          cx="8"
          cy="8"
          fill="none"
          r="7"
          stroke-dasharray="43.982297150257104"
          stroke-dashoffset="17.59291886010284"
          stroke-linecap="round"
          stroke-width="2"
          transform="rotate(-90 8 8)"
        />
      </svg>
      <span
        class="stroke-green-600 dark:stroke-green-400 text-green-600 dark:text-green-400"
      >
        60
        %
      </span>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`ProgressBadge styles should render the component correctly 4`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="flex items-center justify-center gap-1 rounded-md border border-neutral-200 px-1 py-0.5 font-semibold text-neutral-600 shadow-sm dark:border-neutral-700 dark:text-white/80"
      >
        <svg
          height="16"
          width="16"
        >
          <circle
            class="stroke-gray-200 dark:stroke-neutral-600"
            cx="8"
            cy="8"
            fill="none"
            r="7"
            stroke-width="2"
          />
          <circle
            class="stroke-primary dark:stroke-white text-neutral-600 dark:text-white/80"
            cx="8"
            cy="8"
            fill="none"
            r="7"
            stroke-dasharray="43.982297150257104"
            stroke-dashoffset="43.54247417875453"
            stroke-linecap="round"
            stroke-width="2"
            transform="rotate(-90 8 8)"
          />
        </svg>
        <span
          class="stroke-primary dark:stroke-white text-neutral-600 dark:text-white/80"
        >
          0
          %
        </span>
      </div>
    </div>
    <div>
      <div
        class="flex items-center justify-center gap-1 rounded-md border border-neutral-200 px-1 py-0.5 font-semibold text-neutral-600 shadow-sm dark:border-neutral-700 dark:text-white/80"
      >
        <svg
          height="16"
          width="16"
        >
          <circle
            class="stroke-gray-200 dark:stroke-neutral-600"
            cx="8"
            cy="8"
            fill="none"
            r="7"
            stroke-width="2"
          />
          <circle
            class="stroke-amber-600 dark:stroke-amber-400 text-amber-700 dark:text-amber-400"
            cx="8"
            cy="8"
            fill="none"
            r="7"
            stroke-dasharray="43.982297150257104"
            stroke-dashoffset="32.98672286269283"
            stroke-linecap="round"
            stroke-width="2"
            transform="rotate(-90 8 8)"
          />
        </svg>
        <span
          class="stroke-amber-600 dark:stroke-amber-400 text-amber-700 dark:text-amber-400"
        >
          25
          %
        </span>
      </div>
    </div>
    <div>
      <div
        class="flex items-center justify-center gap-1 rounded-md border border-neutral-200 px-1 py-0.5 font-semibold text-neutral-600 shadow-sm dark:border-neutral-700 dark:text-white/80"
      >
        <svg
          height="16"
          width="16"
        >
          <circle
            class="stroke-gray-200 dark:stroke-neutral-600"
            cx="8"
            cy="8"
            fill="none"
            r="7"
            stroke-width="2"
          />
          <circle
            class="stroke-green-600 dark:stroke-green-400 text-green-600 dark:text-green-400"
            cx="8"
            cy="8"
            fill="none"
            r="7"
            stroke-dasharray="43.982297150257104"
            stroke-dashoffset="17.59291886010284"
            stroke-linecap="round"
            stroke-width="2"
            transform="rotate(-90 8 8)"
          />
        </svg>
        <span
          class="stroke-green-600 dark:stroke-green-400 text-green-600 dark:text-green-400"
        >
          60
          %
        </span>
      </div>
    </div>
    <div>
      <div
        class="flex items-center justify-center gap-1 rounded-md border border-neutral-200 px-1 py-0.5 font-semibold text-neutral-600 shadow-sm dark:border-neutral-700 dark:text-white/80"
      >
        <svg
          height="16"
          width="16"
        >
          <circle
            class="stroke-gray-200 dark:stroke-neutral-600"
            cx="8"
            cy="8"
            fill="none"
            r="7"
            stroke-width="2"
          />
          <circle
            class="stroke-green-700 dark:stroke-green-500 text-green-700 dark:text-green-500"
            cx="8"
            cy="8"
            fill="none"
            r="7"
            stroke-dasharray="43.982297150257104"
            stroke-dashoffset="4.3982297150257095"
            stroke-linecap="round"
            stroke-width="2"
            transform="rotate(-90 8 8)"
          />
        </svg>
        <span
          class="stroke-green-700 dark:stroke-green-500 text-green-700 dark:text-green-500"
        >
          90
          %
        </span>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="flex items-center justify-center gap-1 rounded-md border border-neutral-200 px-1 py-0.5 font-semibold text-neutral-600 shadow-sm dark:border-neutral-700 dark:text-white/80"
    >
      <svg
        height="16"
        width="16"
      >
        <circle
          class="stroke-gray-200 dark:stroke-neutral-600"
          cx="8"
          cy="8"
          fill="none"
          r="7"
          stroke-width="2"
        />
        <circle
          class="stroke-green-700 dark:stroke-green-500 text-green-700 dark:text-green-500"
          cx="8"
          cy="8"
          fill="none"
          r="7"
          stroke-dasharray="43.982297150257104"
          stroke-dashoffset="4.3982297150257095"
          stroke-linecap="round"
          stroke-width="2"
          transform="rotate(-90 8 8)"
        />
      </svg>
      <span
        class="stroke-green-700 dark:stroke-green-500 text-green-700 dark:text-green-500"
      >
        90
        %
      </span>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
